SendLog.py

```diff
@@ -126,90 +220,576 @@ def event_stream(xcos_file_id):
+
+
+				log_file = open(log_dir + log_name, "r+")
+
+				if not ( line.set(get_line_and_state(log_file,line_count)) or line.get_state() != ENDING or len(figure_list) > 0 ):
+				    break
+
+
+				if line.get_state()== BLOCK_IDENTIFICATION:
+				    # Split the line obtained from log file and the 8th element is line ID
+				    logLine = line.get_line()
+
+				    line_contents = logLine.split(' ')
+				    # Checking if the current line ID is same as first line ID and is the first occurence of matching
+				    if(line_id == line_contents[7] and not put_delay):
+					delay_length = line_count - 1 # The count of total number of lines in the chart
+					put_delay = True
+				    # The first line ID 
+				    if(line_count == 1):
+					line_id = line_contents[7]
+				    yield "event: block\ndata: "+logLine+"\n\n"
+
+
+				elif line.get_state() != DATA:
+				    gevent.sleep(LOOK_DELAY)    
+
+
+
+				else:
+				    # Split the line obtained from log file and the 8th element is line ID
+				    logLine = line.get_line()
+
+				    line_contents = logLine.split(' ')
+				    # Checking if the current line ID is same as first line ID and is the first occurence of matching
+				    if(line_id == line_contents[7] and not put_delay):
+					delay_length = line_count - 1 # The count of total number of lines in the chart
+					put_delay = True
+
+				    # The first line ID 
+				    if(line_count == 1):
+					line_id = line_contents[7]
+				    yield "event: log\ndata: "+logLine+ "\n\n"
+
+
+				# Reset line, so server won't send same line twice
+				line = line_and_state(None, NOLINE)
+				line_count = line_count + 1
+				log_file.close()
+			#webbrowser.open_new_tab("images/img_test.png")#this displays saved image in a new window
+			# Finished Sending Log
+			kill_scilab()
+			# Notify Client
+			yield "event: DONE\ndata: None\n\n"
+
+		# For processes taking more than 10 seconds
+		except subprocess.TimeoutExpired:
+		    	# Check for errors in Scilab  modified_shank
+			if "Empty diagram" in scilab_out:
+				yield "event: ERROR\ndata: Empty diagram\n\n"
+				kill_scilab()
+				return
+	 
+
+			# Open the log file
+			log_file = open(log_dir + log_name, "r")
+
+			#identify block using file
+			#identify_block= open(identify_block_name, "r")
+			#block_value=identify_block.readline()
+			#identify_block.close()
+			#yield "event: block\ndata: " + block_value + "\n\n"
+			# Check for empty diagram
+	
+			# Start sending log
+			line = line_and_state(None, NOLINE)
+			while (1):
+
+				# The chart must be updated for all the various lines it has in 0.1 seconds.
+				# Hence dividing the sleep time 0.1 by the number of lines the chart contains
+
+				if put_delay:
+				    gevent.sleep(0.1 / delay_length)
+
+
+				log_file = open(log_dir + log_name, "r+")
+
+				if not ( line.set(get_line_and_state(log_file,line_count)) or line.get_state() != ENDING or len(figure_list) > 0 ):
+				    break
+
+
+				if line.get_state()== BLOCK_IDENTIFICATION:
```

Comment:
A lot of code has been repeated in the ```try``` as well as the ```except``` block. Please figure out a way to remove excess repetitive code